DQM/HcalCommon/src/ElectronicsQuantity.cc

```diff
@@ -6,41 +6,106 @@ namespace hcaldqm
 	{
 		int getValue_FED(HcalElectronicsId const& eid)
 		{
-			int v = 0;
-			if (eid.isVMEid())
-				v = eid.dccid();
-			else
-			  v = utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN + 
-					constants::FED_VME_NUM;
-			return v;
+			//int v = 0;
+			//if (eid.isVMEid())
+			//	v = eid.dccid();
+			//else
+			//  v = utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN + 
+			//		constants::FED_VME_NUM;
+			//return v;
+			unsigned int fed = utilities::crate2fed(eid.crateId(), eid.slot());
+			int index = -1;
+			auto it = fedList.begin();
+			for (; it != fedList.end(); ++it) {
+				if (*it == fed) {
+					index = std::distance(fedList.begin(), it);
+					break;
+				}
+			}
+			return index;
 		}
 
 		int getValue_FEDuTCA(HcalElectronicsId const& eid)
 		{
-		  return utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN;
+		  //return utilities::crate2fed(eid.crateId(),eid.slot());
+			unsigned int fed = utilities::crate2fed(eid.crateId(), eid.slot());
+			int index = -1;
```

Comment:
OK, changed to this way for six FED and crate bin lookup functions (getValue_FED, getValue_FEDuTCA, getValue_FEDVME, getValue_Crate, getValue_CrateuTCA, getValue_CrateVME).